app/src/main/java/org/exoplatform/activity/BoardingActivity.java

```diff
@@ -203,35 +211,86 @@ public void statusBarColor(){
-            String url = getIntent().getStringExtra(INTENT_KEY_URL);
-            if(url != null && !url.equals("")) {
-                openWebViewWithURL(url);
-            } else {
-                isFromInstances = getIntent().getBooleanExtra("isFromInstance",false);
-                if (!isFromInstances) {
-                    openWebViewWithURL(serverToConnect.getUrl().toString());
+        try {
+            setWakeUpActivityRoot(new ResultHandler<Boolean>() {
+                @Override
+                public void onSuccess(Boolean isSessionsAlive) {
+                     if (isSessionsAlive) {
+                         String url = getIntent().getStringExtra(INTENT_KEY_URL);
+                         if(url != null && !url.equals("")) {
+                             openWebViewWithURL(url);
+                         } else {
+                             isFromInstances = getIntent().getBooleanExtra("isFromInstance",false);
+                             if (!isFromInstances) {
+                                 openWebViewWithURL(serverToConnect.getUrl().toString());
+                             }
+                         }
+                     }else{
+                         isFromInstances = getIntent().getBooleanExtra("isFromInstance",false);
+                         if (!isFromInstances) {
+                             Intent intent = new Intent(BoardingActivity.this, ConnectToExoListActivity.class);
+                             startActivity(intent);
+                         }
+                     }
                 }
-            }
-        }
+                @Override
+                public void onFailure(Exception e) {
 
-        if (BuildConfig.DEBUG) {
-            long minSinceLastVisit = (System.nanoTime() - lastVisit) / (60000000000L);
-            Log.d(this.getClass().getName(), "*** Minutes since last visit : " + minSinceLastVisit);
+                }
+            });
+        } catch (IOException e) {
+            Log.e("error", String.valueOf(e));
         }
     }
 
     private void openWebViewWithURL(String url) {
         if (url == null)
             throw new IllegalArgumentException("URL must not be null");
-
         Intent intent = new Intent(this, WebViewActivity.class);
         intent.putExtra(INTENT_KEY_URL, url);
         this.startActivity(intent);
     }
+
+    private void setWakeUpActivityRoot(final ResultHandler<Boolean> handler) throws IOException {
+        SharedPreferences prefs = App.Preferences.get(this);
+        Server serverToConnect = new ServerManagerImpl(prefs).getLastVisitedServer();
+        String username = prefs.getString("connectedUsername","username");
+        String cookies = prefs.getString("connectedCookies","cookies");
+        if (serverToConnect != null) {
+            final String url = serverToConnect.getUrl().getProtocol() + "://" + serverToConnect.getShortUrl() + "/portal/rest/state/status/" + username;
```

Comment:
COuld you put this in a global variable to simplify changing it if needed

Reply:
Done